bw2data/project.py

```diff
@@ -52,13 +56,52 @@ def __lt__(self, other):
         else:
             return self.name.lower() < other.name.lower()
 
+    def _get_base_directories(self):
+        envvar = maybe_path(os.getenv("BRIGHTWAY2_DIR"))
+        if envvar:
+            if not envvar.is_dir():
+                raise OSError(
+                    (
+                        "BRIGHTWAY2_DIR variable is {}, but this is not"
+                        " a valid directory"
+                    ).format(envvar)
+                )
+            else:
+                print(
+                    "Using environment variable BRIGHTWAY2_DIR for data "
+                    "directory:\n{}".format(envvar)
+                )
+                return envvar.absolute()
+
+        return Path(PlatformDirs("Brightway3", "pylca").user_data_dir)
+
+    @property
+    def dir(self):
+        return Path(self._get_base_directories()) / safe_filename(
+            self.name, full=self.full_hash
+        )
+
+    def set_sourced(self) -> None:
+        """Set the project to be event sourced."""
+        self.is_sourced = True
+        self.revision = uuid.uuid4()
+        self.save()
+
+    def add_revision(self, patch: str, revision: str | None = None) -> None:
```

Comment:
```suggestion
    def add_revision(self, patch: DeepDiff, revision: str | None = None) -> None:
```

This would be another possibility, we have to decide who has the responsibility of applying future patches. If it's the `Project`, then there's a good chance it should know about `DeepDiff` instead just `str` to files…